decoder/src/lib.rs

```diff
@@ -623,26 +675,32 @@ impl<'t, 'b> Decoder<'t, 'b> {
                     args.push(Arg::F32(f32::from_bits(data)));
                 }
                 Type::BitField(range) => {
-                    let data: u64;
-
-                    match range.end {
-                        0..=8 => {
+                    let mut data: u64;
+                    let lowest_byte = range.start / 8;
+                    // -1 because `range` is range-exclusive
+                    let highest_byte = (range.end - 1) / 8;
+                    let truncated_sz = highest_byte - lowest_byte + 1; // in octets
```

Comment:
I think `size_after_truncation` (if truncation is an actual word) would be a clearer name.

Reply:
is an actual word; renamed